### PR TITLE
Improve Sensu edition header handling

### PR DESCRIPTION
**Pull request by palourde**
## What is this change?

It inspects the headers returned by the Sensu API on each request to verify the Sensu edition, and updates sensuctl configuration if required.

## Why is this change necessary?

It ensures a user has access to the enterprise version of sensuctl whenever they upgrade Sensu Backend.

## Does your change need a Changelog entry?

Added

## Do you need clarification on anything?

Nope

## Were there any complications while making this change?

Nope

## Have you reviewed and updated the documentation for this change? Is new documentation required?

Nope

### CODE REVIEW

File: cli/client/client.go [hunk 1]
```diff
@@ -113,6 +114,23 @@ func New(config config.Config) *RestClient {
 		return nil
 	})
 
+	// Verify the Sensu edition and update the sensuctl configuration if required
+	restyInst.OnAfterResponse(func(c *resty.Client, resp *resty.Response) error {
+		// Retrieve the Sensu edition from the response header
+		headerEdition := resp.Header().Get(types.EditionHeader)
+		if headerEdition == "" {
```
Comment (nikkiki) [inline cli/client/client.go]: Will there ever be a case that we don't return an edition header?

Comment (palourde) [reply cli/client/client.go]: I could see the case where someones run a recent sensuctl binary with an older sensu-backend.

**Comment by palourde** [comment]
@nikkiki Indeed, showing the current edition with `sensuctl config view` would be useful!

File: cli/client/client.go (hunk 1 above)
Comment (nikkiki) [reply cli/client/client.go]: Ah good call!

**Comment by palourde** [comment]
@nikkiki `config view` subcommand has been updated!

**Pull request merged**